src/main/kotlin/org/rust/ide/navigation/goto/RsCurrentActionService.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.navigation.goto
+
+import com.intellij.codeInsight.navigation.actions.GotoDeclarationAction
+import com.intellij.codeInsight.navigation.actions.GotoDeclarationOnlyAction
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.actionSystem.ex.AnActionListener
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+
+@Service
+class RsCurrentActionService {
+    var isGoToDeclarationAction: Boolean = false
```

Comment:
Let's add some synchronization for this property. As I can see, we use it from different threads and call setter not under write action